_posts/2021-07-01-origami-on-npm-and-how-to-migrate.md

```diff
@@ -0,0 +1,107 @@
+- Services such as Snyk, Dependabot, and Renovate will be able to work for the Origami dependencies a project has.
+
+# What has changed for Origami users?
+
+The code for Origami components has remained largely the same, aiming to make the migration simpler for our users. The one change we had to make was making the package name for all components now start with `@financial-times/`, as they are published under the financial-times namespace on npm. This change means that when importing the javascript and/or sass of a component, you will need to prepend `@financial-times/` to the imported package name.
+
+If using Origami components via NPM, it is required to use NPM version 7, this is because version 7 ensures that Origami components are only installed once and no conflicting versions are installed. Origami components do not support using older versions of NPM.
+
+# Plans for the future
+
+Origami used to be versioned only with a major version (Origami 1), this made it hard to track any of the changes being made to Origami which did not warrant incrementing the major version. Now we have incremented the major version, we thought it was a good time to include minors in the versioning scheme. This will make it simpler to make more frequent changes to Origami as a whole and have a record of what thoses changes were and when they were made.
+
+We are looking for people's thoughts on what Origami is missing and/or how Origami could be reimagined to help better achieve it's aim to unify and document the style and experience across the digital products of every FT brand. Some of the parts of Origami we are wanting to improve are around how the documentation and demonstrations for the components are built. Currently they are built in isolation of one another, we would like to see if building them together would lead to a better documented and test set of components.
+
+# First npm-only Origami Component
+
+We've been working on an autocomplete Origami component, which has been requested by several different FT products. By making this an npm-only component, we've been able to benefit from autocomplete packages which already exist on NPM, such as the one made by Gov UK named [accessible-autocomplete](https://github.com/alphagov/accessible-autocomplete). This has greatly reduced the amount of code we would have had to write ourselves and meant we could create a prototype for products to trial in a shorter turnaround then previously possible.
+
+# Migration Guides
+
+## Using Origami via Bower
+
+All Origami components are published to the `@financial-times` namespace of NPM.
+
+1. Run the project locally in its current state, using the version of NPM the project expects right now
+1. Upgrade to the latest major versions of any Origami bower components you are using:
+    1. Confirm the project is still working correctly
+1. If the project is not using NPM version 7:
```

Comment:
We found that npm 6 does work, we should update these guides to mention how to get npm 6 working if the project can't upgrade to npm 7.

This is the commit in polyfill.io that got npm 6 working correctly -- https://github.com/Financial-Times/polyfill-service/pull/2616/commits/8b9c644481c488c1a3ee5efd372e9c21408bc821

Should we wrap that script into something which projects can install and depend on rather than copy-pasting it?

Reply:
Update: npm 6 could work but it is not possible to do it in an automated way. I had a chat with chee about npm 6 and we decided to not document npm 6 as we do not provide support for it